test/state/host.cpp

```diff
@@ -271,13 +271,22 @@ evmc::Result Host::create(const evmc_message& msg) noexcept
                                           evmc::Result{EVMC_FAILURE};
     }
 
-    if (m_rev >= EVMC_PRAGUE && (is_eof_container(initcode) || is_eof_container(code)))
+    if (!code.empty() && code[0] == 0xEF)
     {
-        if (validate_eof(m_rev, code) != EOFValidationError::success)
+        if (m_rev >= EVMC_LONDON && m_rev < EVMC_PRAGUE)
+        {
+            // EIP-3541: Reject EF code.
             return evmc::Result{EVMC_CONTRACT_VALIDATION_FAILURE};
+        }
+        else if (m_rev >= EVMC_PRAGUE)
```

Comment:
If you reorder `m_rev` checks they should simplify.

Reply:
Not sure what you mean. Wihout CREATE3/4 it's a failure in all forks, so we could just drop this commit from this PR and get to pre-EOF logic
```cpp
      if (m_rev >= EVMC_LONDON && !code.empty() && code[0] == 0xEF)  // Reject EF code.
        return evmc::Result{EVMC_CONTRACT_VALIDATION_FAILURE};
```
Then this commit will still be part of #553